### PR TITLE
v1 of the plugin

### PR DESCRIPTION
**Pull request by ashb**
This plugin operates as follows:

- It uses `functools.wraps` to "hook" in to various places in Airflow
  that don't have pre-existing hook points.

  This does mean that if those function names change in the future this
  will stop working.

- It runs a thread in the scheduler process to make requests to the
  update server.

  It hooks in to airflow.jobs.scheduler_job.SchedulerJob._execute_helper
  to start the thread.

- Available updates are stored in the database for the UI to query

- The UI display is done by customizing the `appbuilder.base_template`,
  as again, there aren't enough hook points to inject our front end code
  any other way. This needs AIRFLOW-6733 and AIRFLOW-6734 to work.

- This notice is shown for all (logged in) users, but the dismiss button is
  only visible for admins.

- There aren't any tests right now :(

### CODE REVIEW

**Comment by ashb** [comment]
Screenshots of what this looks like in the UI:

image

image

The "Ignore this update" will only be visible to admins right now (technically anyone with the "can_dismiss on UpdateAvailableView" permission - which admins get automatically)

File: astronomer/airflow/version_check/models.py [hunk 1]
```diff
@@ -0,0 +1,94 @@
...
+        """
+        Ensure that the singleton row exists in this table
+        """
+        with create_session() as session:
+            try:
+                session.bulk_save_objects([cls(singleton=True)])
+            except sqlalchemy.exc.IntegrityError:
+                # Already exists, we're good
+                pass
+                session.rollback()
```
Comment (andriisoldatenko) [inline astronomer/airflow/version_check/models.py]: @ashb why do we need pass here??

Comment (ashb) [reply astronomer/airflow/version_check/models.py]: We don't :)

**Comment by andriisoldatenko** [comment]
@ashb in general LGTM, tests would be nice to have :)

File: astronomer/airflow/version_check/update_checks.py [hunk 2]
```diff
@@ -0,0 +1,264 @@
...
+            )
+
+    def _make_fake_response(self):
+        v = version.parse(AIRFLOW_VERSION)
+
+        new_version = f'{v.major}.{v.minor}.{v.micro+1}-1'
+
+        return {
+            'format_version': '1.0',
+            'available_versions': [
```
Comment (andriisoldatenko) [inline astronomer/airflow/version_check/update_checks.py]: IMO better to rename:
`available_versions ` -> `available_releases`

**Comment by ashb** [comment]
Now with bearest of minimal tests.

(Checks that the plugin loads, and that the template hackery I've added has worked)

Need to add some tests against the HTTP checker part then this is done.

**Comment by ashb** [comment]
Oh and I need to change that message.

**Comment by ashb** [comment]
The unit tests should probably be extneded, but good enough for now.

**Pull request merged**